Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -815,7 +815,8 @@ namespace AzToolsFramework
 
             if (templateRef.has_value())
             {
-                return templateRef->get().IsDirty();
+                return !templateRef->get().IsProcedural() && // all procedural prefabs are read-only
+                        templateRef->get().IsDirty();
```

Comment:
This is a false equivalency. A template of a procedural prefab can absolutely be modified in memory through an API call since there aren't any guardrails against that. We should either put those guardrails in (may not be easily done) or we should change the logic somewhere higher up the hierarchy to say 'if the template is dirty and it is a procedural prefab, then ignore whatever action you are about to perform'